Refactor Message class.

o01eg opened the pull request:
1. Remove sender_id and receiver_id from header because we know this data in most cases from player connection.
2. Merge PLAYER_CHAT and LOBBY_CHAT messages.
3. Send broadcast Message by ServerNetworking::SendMessageAll and other
Message by PlayerConnection::SendMessage.
4. Add player_id into Message body where it was used in header.

o01eg committed:
Clean up unused player_id.

LGM-Doyle reviewed (commented):
@Vezzra this is straightforward enough that it either works or it does not.  We have not branched v0.4.7 or officially started testing.  I think that any problems with this will show up in testing and benching it for weeks is unwarranted.

However, I bow to @geoffthemedio's greater experience.

@o01eg the code is good, clear and easy to understand.

You explained what the changes that you made are, but you did not explain why you made them?

Why is this better?

o01eg committed:
Remove do-nothing Exception handler. Specify boost::bad_lexical_cast exception handler.

o01eg committed:
Clean up unused text variable.

o01eg committed:
Change error messages for boost::bad_lexical_cast.

o01eg committed:
Use enum Parts instead magic values.

o01eg committed:
Clean up unused ACKNOWLEDGEMENT.

o01eg committed:
Clean up unused ServerNetworking::SendMessage.

o01eg committed:
Refactor Message class.

1. Remove sender_id and receiver_id from header as we know this data in most cases from connection.
2. Merge PLAYER_CHAT and LOBBY_CHAT messages.
3. Send broadcast Message by ServerNetworking::SendMessageAll and other
Message by PlayerConnection::SendMessage.
4. Add player_id into Message body where it was used in header.

o01eg committed:
Clean up Message named ctors arguments.

LGM-Doyle reviewed (commented):
I've played a few hundred turns with this PR and observed no problems.

I've noted two small changes.  

After those are fixed we should merge this quickly because it touches on a lot of code and it will often need to be rebased as other code is merged.  Plus @o01eg is a newer contributor. 😄

o01eg committed:
Catch boost::bad_lexical_cast when parse text to int.
